cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -176,58 +193,37 @@ public void createIfNotExists(ProfileId profileId, Profile profile) {
-      Profile profile = profileDataset.getProfile(profileId);
-      if (profile == null) {
-        throw new NotFoundException(profileId);
-      }
-
-      // The profile status must be DISABLED
-      if (profile.getStatus() == ProfileStatus.ENABLED) {
-        throw new ProfileConflictException(
-          String.format("Profile %s in namespace %s is currently enabled. A profile can " +
-                          "only be deleted if it is disabled", profileId.getProfile(), profileId.getNamespace()),
-          profileId);
-      }
-
-      // There must be no assigments to this profile
-      Set<EntityId> assignments = profileDataset.getProfileAssignments(profileId);
-      if (!assignments.isEmpty()) {
-        throw new ProfileConflictException(
-          String.format("This profile %s is still assigned to %d entities, like %s. " +
-                          "Please delete all assignments before deleting the profile.",
-                        profileId.toString(), assignments.size(), assignments.iterator().next()),
-          profileId);
-      }
+      Profile profile = getProfileDataset(context).getProfile(profileId);
+      deleteProfile(context, profileId, profile);
+    }, NotFoundException.class, ProfileConflictException.class);
+  }
 
-      // There must be no running programs using the profile
-      AppMetadataStore appMetadataStore = AppMetadataStore.create(cConf, context, datasetFramework);
-      Map<ProgramRunId, RunRecordMeta> activeRuns;
-      Predicate<RunRecordMeta> runRecordMetaPredicate = runRecordMeta -> {
-        // the profile comes in system arguments with the scoped name
-        String scopedName = runRecordMeta.getSystemArgs().get(SystemArguments.PROFILE_NAME);
-        return scopedName != null && scopedName.equals(profileId.getScopedName());
-      };
-      if (profileId.getNamespaceId().equals(NamespaceId.SYSTEM)) {
-        activeRuns = appMetadataStore.getActiveRuns(runRecordMetaPredicate);
-      } else {
-        activeRuns = appMetadataStore.getActiveRuns(Collections.singleton(profileId.getNamespaceId()),
-                                                    runRecordMetaPredicate);
-      }
-      if (!activeRuns.isEmpty()) {
-        throw new ProfileConflictException(
-          String.format("The profile %s is in use by %d active runs, like %s. Please stop all active runs " +
-                          "before deleting the profile.",
-                        profileId.toString(), activeRuns.size(), activeRuns.keySet().iterator().next()),
-          profileId);
+  /**
+   * Delete all profiles in a given namespace. Deleting all profiles in SYSTEM namespace is not allowed.
+   *
+   * @param namespaceId the id of the namespace
+   */
+  public void deleteAllProfiles(NamespaceId namespaceId)
+    throws MethodNotAllowedException, NotFoundException, ProfileConflictException {
+    if (namespaceId.equals(NamespaceId.SYSTEM)) {
+      throw new MethodNotAllowedException("");
```

Comment:
Should have a message. "Deleting all system profiles is not allowed".

Reply:
Added.